pkg/grpc/auth.go

```diff
@@ -0,0 +1,27 @@
+package grpc
+
+import (
+	"context"
+	"net/http"
+
+	v1 "github.com/dapr/dapr/pkg/messaging/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+func setAPIAuthenticationMiddlewareUnary(apiToken, authHeader string) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		md, _ := metadata.FromIncomingContext(ctx)
```

Comment:
I think we need to check the error. if it has error, it needs to return proper error.

Reply:
I'm not sure what you mean here. What error?

Reply:
md, ok := metadata.FromIncomingContext(ctx)

Correction: checking ok value.

Reply:
ok

Reply:
fixed